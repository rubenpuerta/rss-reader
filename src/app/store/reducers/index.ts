export * from './app.reducer';
export * from './ui.reducer';
