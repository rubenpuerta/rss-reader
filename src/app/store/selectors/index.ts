export * from './rss.selectors';
export * from './ui.selectors';
