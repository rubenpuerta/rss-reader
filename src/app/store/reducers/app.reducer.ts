import { InjectionToken } from '@angular/core';
import { ActionReducerMap } from '@ngrx/store';

import { rssReducer, RssState } from '@store/reducers/rss.reducer';
import { CommonUIState, uiReducer } from '@store/reducers/ui.reducer';

export interface AppState {
	rssItems: RssState;
	commonUI: CommonUIState;
}

export const appReducer = new InjectionToken<ActionReducerMap<AppState>>('App reducers token', {
	factory: () => ({
		rssItems: rssReducer,
		commonUI: uiReducer
	})
});
