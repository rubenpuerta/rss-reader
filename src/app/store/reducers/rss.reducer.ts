import { createReducer, on } from '@ngrx/store';

import { NewsItem } from '@models/rss-news.model';
import { DataLoadActions } from '@store/actions/data-load.actions';

export interface RssState {
	news: NewsItem[];
}

export const rssInitialState: RssState = {
	news: []
};

export const rssReducer = createReducer(
	rssInitialState,
	on(DataLoadActions.initialDataLoadSuccess, (state, { news }) => ({
		...state,
		news
	}))
);
