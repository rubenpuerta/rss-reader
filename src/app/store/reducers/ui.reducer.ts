import { createReducer, on } from '@ngrx/store';

import { UIActions } from '@store/actions';

export interface CommonUIState {
	isLoading: boolean;
}

export const commonUIInitialState: CommonUIState = {
	isLoading: false
};

export const uiReducer = createReducer(
	commonUIInitialState,
	on(UIActions.startLoading, (state) => ({ ...state, isLoading: true })),
	on(UIActions.stopLoading, (state) => ({ ...state, isLoading: false }))
);
