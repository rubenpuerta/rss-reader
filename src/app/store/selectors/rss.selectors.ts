import { createFeatureSelector, createSelector } from '@ngrx/store';

import { RssState } from '@store/reducers/rss.reducer';

const getRssItemsState = createFeatureSelector<RssState>('rssItems');

export const getRssFeed = createSelector(getRssItemsState, ({ news }) => news);
