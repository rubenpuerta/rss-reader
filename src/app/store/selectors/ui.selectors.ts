import { createFeatureSelector, createSelector } from '@ngrx/store';

import { CommonUIState } from '@store/reducers';

const getCommonUIState = createFeatureSelector<CommonUIState>('commonUI');

export const getIsAppLoading = createSelector(getCommonUIState, (uiState) => uiState?.isLoading);
