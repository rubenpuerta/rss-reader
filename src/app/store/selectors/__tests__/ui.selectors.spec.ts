import { commonUIInitialState, CommonUIState } from '@store/reducers';
import { getIsAppLoading } from '@store/selectors';

const getCommonUIState = (newValue?: Partial<CommonUIState>) => ({
	commonUI: {
		...commonUIInitialState,
		...newValue
	}
});

describe('UI Selectors', () => {
	it('should return UI state', () => {
		const result = getCommonUIState();
		expect(result).toMatchSnapshot();
	});

	it('should return isLoading false', () => {
		const result = getIsAppLoading(getCommonUIState());
		expect(result).toMatchSnapshot();
	});

	it('should return isLoading true', () => {
		const result = getIsAppLoading(getCommonUIState({ isLoading: true }));
		expect(result).toMatchSnapshot();
	});
});
