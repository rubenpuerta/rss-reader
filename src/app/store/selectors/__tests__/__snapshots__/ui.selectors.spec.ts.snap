// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Selectors should return UI state 1`] = `
Object {
  "commonUI": Object {
    "isLoading": false,
  },
}
`;

exports[`UI Selectors should return isLoading false 1`] = `false`;

exports[`UI Selectors should return isLoading true 1`] = `true`;
